lib/xml.js

```diff
@@ -0,0 +1,419 @@
+// adapted from: https://raw.githubusercontent.com/mckamey/jsonml/master/jsonml-xml.js
+/* eslint-disable no-use-before-define */
+/* global ActiveXObject */
+'use strict';
+
+/*
+  jsonml-xml.js
+  JsonML XML utilities
+
+  Created: 2007-02-15-2235
+  Modified: 2012-11-03-2051
+
+  Copyright (c)2006-2012 Stephen M. McKamey
+  Distributed under The MIT License: http://jsonml.org/license
+
+  This file creates a global JsonML object containing these methods:
+
+    toXML(string|array, filter)
+      Converts JsonML to XML nodes
+
+    toXMLText(JsonML, filter)
+      Converts JsonML to XML text
+
+    fromXML(node, filter)
+      Converts XML nodes to JsonML
+
+    fromXMLText(xmlText, filter)
+      Converts XML text to JsonML
+*/
+
+const { JSDOM } = require('jsdom');
+
+const win = typeof window === 'undefined' ?
+  (new JSDOM()).window :
+  window; // eslint-disable-line no-undef
+// enable usage of XML DOM, fallback to HTML DOM
+const doc = parseXML('<xml/>') || win.document;
+
+module.exports = {
+  fromXML,
+  fromXMLText,
+  isXML,
+  onerror: null,
+  parseXML,
+  renderXML,
+  toXML,
+  toXMLText,
+};
+
+/**
+  * Determines if the value is an Array
+  *
+  * @private
+  * @param {*} val the object being tested
+  * @return {boolean} true if an Array
+  */
+const isArray = Array.isArray || function(val) {
+  return (val instanceof Array);
+};
+
+/**
+  * Creates a DOM element
+  *
+  * @private
+  * @param {string} tag The element's tag name
+
+  * @return {Node} DOM Element
+  */
+const createElement = function(tag) {
+  if (!tag) {
+    // create a document fragment to hold multiple-root elements
+    if (doc.createDocumentFragment) {
+      return doc.createDocumentFragment();
+    }
+
+    tag = '';
+
+  } else if (tag.charAt(0) === '!') {
+    return doc.createComment(tag === '!' ? '' : tag.substr(1) + ' ');
+  }
+
+  return doc.createElement(tag);
+};
+
+/**
+  * Appends an attribute to an element
+  *
+  * @private
+  * @param {Node} elem The element
+  * @param {Object} attr Attributes object
+  * @return {Node} DOM Element
+  */
+const addAttributes = function(elem, attr) {
+  // for each attributeName
+  for (const name in attr) {
+    if (attr.hasOwnProperty(name)) {
+      // attributes
+      elem.setAttribute(name, attr[name]);
+    }
+  }
+  return elem;
+};
+
+/**
+  * Appends a child to an element
+  *
+  * @private
+  * @param {Node} elem The parent element
+  * @param {Node} child The child
+  */
+const appendDOM = function(elem, child) {
+  if (child) {
+    if (elem.nodeType === 8) { // comment
+      if (child.nodeType === 3) { // text node
+        elem.nodeValue += child.nodeValue;
+      }
+
+    } else if (elem.canHaveChildren !== false) {
+      elem.appendChild(child);
+    }
+  }
+};
+
+/**
+  * Default error handler
+  * @param {Error} ex - error
+  * @return {Node} Text node
+  */
+const onError = function(ex) {
+  return doc.createTextNode('[' + ex + ']');
```

Comment:
Since we're updating this for stylistic stuff, might be worth including template strings in some of these instances, like here

Reply:
I tried to limit the amount of lint changes being applied other than `var` and exports, but yes I'd be ok with that!